cache.go

```diff
@@ -552,6 +552,20 @@ func (c *Cache[K, V]) OnEviction(fn func(context.Context, EvictionReason, *Item[
 	}
 }
 
+// Range iterate over all items and calls fn function. It calls fn function
+// until it returns false.
+func (c *Cache[K, V]) Range(fn func(item *Item[K, V]) bool) {
+	c.items.mu.Lock()
+	defer c.items.mu.Unlock()
+
+	for k := range c.items.values {
+		item := c.get(k, false).Value.(*Item[K, V])
+		if !fn(item) {
```

Comment:
It should be possible to execute other cache methods without causing any deadlocks as mentioned in this comment: https://github.com/jellydator/ttlcache/issues/71#issuecomment-1086167465

`sync.Map`'s `Range()` has this documented quite well:
`Range does not block other methods on the receiver; even f itself may call any method on m. `

Reply:
I didn't get the main logic here. I added `mu.Lock()`-`mu.Unlock()` in `Range()`. How can I avoid deadlocks?

Reply:
The problem is that the user might want to use this method as follows:
```go
cache.Range(func(item *Item){
    cache.Delete(item.Key())
    cache.Set(item.Key() + "newkey", "123")
})
```

This is just an example but it illustrates my point: calling a cache method inside the range func will cause a deadlock.

The simplest solution for now would be to lock and unlock the mutex before calling the range func:
```go
func (c *Cache) Range(fn func(*Item)) {
    c.items.mu.RLock()
    for next := c.items.lru.Next(); next != nil; next = c.items.lru.Next() {
        item := next.Value.(*Item)
        c.items.mu.RUnlock()
        
        if !fn(item) {
            return
        }
        
        if c.items.lru.Next() != nil {
            c.items.mu.RLock()
        }
    }
}
```

Reply:
I did some changes. Please let me know if i missed something.